DashAI/front/src/components/ExperimentsTable.jsx

```diff
@@ -1,73 +1,131 @@
 import React from "react";
 import PropTypes from "prop-types";
-import uuid from "react-uuid";
-import { StyledButton } from "../styles/globalComponents";
-import * as S from "../styles/components/ExperimentsTableStyles";
 
-function ExperimentsTable({ rows, removeExperimentFactory }) {
-  if (rows.length > 0) {
-    return (
-      <S.Table hover>
-        <thead>
-          <S.Tr>
-            <S.Th>#</S.Th>
-            <S.Th>Name</S.Th>
-            <S.Th>Created</S.Th>
-            <S.Th>Edited</S.Th>
-            <S.Th>Task</S.Th>
-            <S.Th>Dataset</S.Th>
-            <S.Th>Remove</S.Th>
-          </S.Tr>
-        </thead>
+import { DataGrid, GridActionsCellItem } from "@mui/x-data-grid";
+import DeleteIcon from "@mui/icons-material/Delete";
+import AddIcon from "@mui/icons-material/Add";
+import { Button, Grid, Paper, Typography } from "@mui/material";
 
-        <tbody>
-          {rows.map((key, index) => (
-            <S.Tr key={uuid()}>
-              <S.Td>{index}</S.Td>
-              <S.Td>{key.name}</S.Td>
-              <S.Td>{key.created}</S.Td>
-              <S.Td>{key.edited}</S.Td>
-              <S.Td>{key.taskName}</S.Td>
-              <S.Td>{key.dataset}</S.Td>
-              {/* <S.Td> */}
-              {/*   <StyledButton */}
-              {/*     variant="dark" */}
-              {/*     onClick={renderFormFactory(key.type, index)} */}
-              {/*   > */}
-              {/*     Configure */}
-              {/*   </StyledButton> */}
-              {/* </S.Td> */}
-              <S.Td>
-                <StyledButton
-                  variant="dark"
-                  onClick={removeExperimentFactory(index)}
-                  style={{
-                    verticalAlign: "middle",
-                    backgroundColor: "#F1AE61",
-                  }}
-                >
-                  <img
-                    alt=""
-                    style={{ marginBottom: "100px" }}
-                    src="/images/trash.svg"
-                    width="20"
-                    height="20"
-                  />
-                </StyledButton>
-              </S.Td>
-            </S.Tr>
-          ))}
-        </tbody>
-      </S.Table>
-    );
-  }
-  return <div />;
+function ExperimentsTable({ initialRows, handleNewExperiment }) {
+  const [rows, setRows] = React.useState(initialRows);
+
+  const deleteUser = React.useCallback(
+    (id) => () => {
+      setTimeout(() => {
+        setRows((prevRows) => prevRows.filter((row) => row.id !== id));
+      });
+    },
+    []
+  );
```

Comment:
```suggestion
  const deleteExperiment = React.useCallback(
    (id) => () => {
      setTimeout(() => {
        setRows((prevRows) => prevRows.filter((row) => row.id !== id));
      });
    },
    []
  );
```
I guess endpoint call should be added here as WIP and not only shadow delete?

Reply:
I forgot to change the name, it came this way from the template. 
I will change the name to make the merge correct.
Anyway, the next pr should extract the code from the component requests. In particular, this request should be in `api/experiment.ts`.